doc/newfeat.texi

```diff
@@ -15,6 +15,13 @@ process is attached, ESS now switches automatically to one (prompting
 you for selection if there are several running). Reloading and quitting
 will now work during a debug session or when R is prompting for input
 (for instance after a crash).
+
+@item @ESS{[R]}: New command @code{ess-r-package-use-dir}. It sets the
+working directory of the current process to the current package directory.
+
+@item @ESS{[R]}: Launching or reloading R from a package file now sets
+the working directory to the package path. You can now launch R from
+files within @code{R/} or @code{src/} without worrying about the wd.
 @end itemize
```

Comment:
Actually what I had in mind is something much less intrusive. I thought that proposing package dir on `M-x R` by default should be enough. 

I am pretty sure this new behavior will be confusing to the users. People might want to start in a sub-dir for whatever reason and messing up with their wd in the background seems like a bad design to me.

Reply:
It's not in background, there's a visible `setwd()` that appears in the terminal. I think we talked past each other, I thought you suggested that behaviour.

I don't think `M-x R` should be prompting anything to the user, that'd be annoying. We could make it an option but I think it should be enabled by default as this looks like good design to me that the working directory is set to the project root by default.

Reply:
I think purposefully setting the wd inside `R/` or `src/` or other subfolders is a very rare use case. I spend my time developing packages and I never needed that. It's always possible to switch to a subfolder after having started R.

Reply:
Sorry, I had completely forgot about `ess-ask-for-ess-directory` because I disabled it years ago. How about this:

- if non-nil, we ask with the package directory as default
- if nil, we switch to the package directory directly (but verbosely)

Reply:
Sounds like a good tradeoff. Also, could you then add a line or two of clarifications to `ess-ask-for-ess-directory`? I am keeping that option on as I find it occasionally useful to be worth one extra RET.